### ISSUE TITLE
GAE bootstrap uses V(s_t) instead of V(s_{t+1}) at non-terminal buffer boundaries

### ISSUE THREAD

**Issue by chrisjz** [SyntheticBrains/nematode#62]
## Summary

`_reservoir_hybrid_base.py` uses `self.last_value` for GAE bootstrapping in `_perform_ppo_update()`, but `self.last_value` is V(s_t) — the value of the state that produced the last action — not V(s_{t+1}) (the successor state). This means non-terminal buffer flushes bootstrap with the wrong value estimate.

## Root Cause

In `run_brain()` (line 536):
```python
self.last_value = value  # V(s_t)
```

Then in `learn()`, after adding the transition to the buffer, if `buffer.is_full()` triggers `_perform_ppo_update()`, the bootstrap value is V(s_t) rather than V(s_{t+1}):

```python
# _perform_ppo_update() line 641-648
last_value = (
    self.last_value  # This is V(s_t), should be V(s_{t+1})
    if self.last_value is not None
    else torch.tensor([0.0], device=self.device)
)
```

The sequence is:
1. `run_brain(s_t)` → sets `self.last_value = V(s_t)`
2. `learn(r_t)` → adds (s_t, a_t, V(s_t), r_t) to buffer → buffer full → PPO update uses V(s_t) as bootstrap
3. But GAE needs V(s_{t+1}) which hasn't been computed yet

Terminal flushes are unaffected since `next_non_terminal` zeros out the bootstrap.

## Impact

Bounded in practice:
- Affects 1 out of every 128 transitions (buffer size), only at mid-episode boundaries
- GAE λ-decay (γλ ≈ 0.95) means the error primarily biases the last few transitions in each rollout segment
- V(s_t) and V(s_{t+1}) are typically close for adjacent states in small-grid environments
- Net effect: slightly noisier advantage estimates at rollout boundaries, well within PPO's robustness margins
- Affects all reservoir-hybrid brains equally (QRH, CRH, CRH-trig) since they share the base class, so ablation comparisons remain valid

## Affected Code

- `packages/quantum-nematode/quantumnematode/brain/arch/_reservoir_hybrid_base.py` — `learn()` and `_perform_ppo_update()`

## Suggested Fix

Defer the PPO update when the buffer fills mid-episode: on the next `run_brain(s_{t+1})` call, compute V(s_{t+1}) in the normal forward pass and then trigger the deferred PPO update with the correct bootstrap. This avoids an extra forward pass through the reservoir + readout.

Alternatively, compute V(s_{t+1}) from the next observation in `learn()`, but this requires a full forward pass (reservoir + feature extraction + critic) which is expensive, especially for quantum reservoirs.
